utils/db_utils.py

```diff
@@ -94,3 +94,18 @@ def query_confirmed_trips(start_date, end_date):
     # logging.debug("After filtering, the actual data is %s" % df.head())
     # logging.debug("After filtering, the actual data is %s" % df.head().trip_start_time_str)
     return df
+
+
+def add_user_stats(user_data, trip_data):
+    trip_df = pd.DataFrame(trip_data)
+    trip_grouped = trip_df.groupby('user_id')
+    for user in user_data:
+        user_id = user['user_id']
+        if user_id in trip_grouped.groups:
+            trips = trip_grouped.get_group(user_id)
+            user['confirmed_trips'] = len(trips)
+            user['first_trip'] = trips['trip_start_time_str'].min()
+            user['last_trip'] = trips['trip_start_time_str'].max()
+            # logging.debug(f"{trip_grouped.get_group(user_id).columns}")
+            # number_of_answered = len(df[df['user_input'] != {}].index) if 'user_input' in df.columns else 0
+    return user_data
```

Comment:
I am not sure that this is the desired implementation.

This may have the desired behavior at this point, but I believe it will no longer do so once https://github.com/e-mission/op-admin-dashboard/issues/51 is implemented

@asiripanich can confirm, but I believe that the goal is for the user table to include the summary of the entire trip history for the user. If we only load a week at a time, this will not have enough information.

Please use `get_first_value_for_field` to get the first and last trips instead - there should be examples of how it is used in the codebase